g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/utils/S3Util.java

```diff
@@ -87,28 +90,30 @@ public Locker(String basePath, SingleComponentDTO compDTO) {
 		}
 
 		public boolean lockFile() {
+			long endTime = System.currentTimeMillis() + waitToLock;
 			String content = Double.toString(random.nextDouble());
-             int time=0;
+
 			do {
-				if (!waitLockfileDisappeared()) {
+				if (!waitLockfileDisappeared(endTime)) {
 					return false;
 				}
 
 				// Write the lock file
-				s3Client.putObject(this.key, content);
-				TimeUtils.sleep(waitS3Operation); // Wait for a while to let S3 finish writing.
+				PutObjectResult pObjResult = s3Client.putObject(this.key, content);
 				// Check file content is correct
+				String putVersionId = pObjResult.getVersionId();
+				String getVersionId = s3Client.getS3Client().getObjectMetadata(config.getBucketName(), this.key).getVersionId();
+				
 				try {
-					if (content.equals(s3Client.readObject(this.key))) {
+					if (putVersionId.equals(getVersionId)) {
```

Comment:
It seems there is no exception from the block. Move it to  line  104?

Reply:
It have exception when the key is null

Reply:
But AmazonS3Exception is caught instead of null pointer exception.

Reply:
It have exception when the key is deleted

Reply:
But AmazonS3Exception is caught instead of null pointer exception.